File: mobile/backend/Luma.Api/Controllers/EntriesController.cs
using Luma.Api.Data;
using Luma.Api.Models;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace Luma.Api.Controllers;

[ApiController]
[Route("api/v1/entries")] 
public class EntriesController : ControllerBase
{
    private readonly AppDbContext _db;

    public EntriesController(AppDbContext db) => _db = db;
    private Guid GetUserId()
    {
        if (Request.Headers.TryGetValue("X-User-Id", out var values) &&
            Guid.TryParse(values.FirstOrDefault(), out var userId))
            return userId;
        return Guid.Parse("11111111-1111-1111-1111-111111111111");
    }

    [HttpPost]
    public async Task<ActionResult> Create(JournalEntry entry)
    {
        entry.UserId = GetUserId();
        entry.Id = Guid.NewGuid();
        entry.CreatedAtUtc = DateTime.UtcNow;

        _db.JournalEntries.Add(entry);
        await _db.SaveChangesAsync();

        return CreatedAtAction(nameof(GetById), new { id = entry.Id }, entry);
    }

    [HttpGet]
    public async Task<ActionResult> GetAll()
    {
        var userId = GetUserId();
        var items = await _db.JournalEntries
            .Where(x => x.UserId == userId)
            .OrderByDescending(x => x.CreatedAtUtc)
            .Take(50)
            .ToListAsync();
        return Ok(items);
    }

    [HttpGet("{id:guid}")]
    public async Task<ActionResult> GetById(Guid id)
    {
        var item = await _db.JournalEntries.FirstOrDefaultAsync(x => x.Id == id && x.UserId == GetUserId());
        return item is null ? NotFound() : Ok(item);
    }

    [HttpPut("{id:guid}")]
    public async Task<ActionResult> Update(Guid id, JournalEntry updated)
    {
        var item = await _db.JournalEntries.FirstOrDefaultAsync(x => x.Id == id && x.UserId == GetUserId());
        if (item is null) return NotFound();

        item.Content = updated.Content;
        item.Mood = updated.Mood;
        item.UpdatedAtUtc = DateTime.UtcNow;

        await _db.SaveChangesAsync();
        return Ok(item);
    }

    [HttpDelete("{id:guid}")]
    public async Task<ActionResult> Delete(Guid id)
    {
        var item = await _db.JournalEntries.FirstOrDefaultAsync(x => x.Id == id && x.UserId == GetUserId());
        if (item is null) return NotFound();

        _db.JournalEntries.Remove(item);
        await _db.SaveChangesAsync();
        return NoContent();
    }
}

File: newBackend/src/Controllers/EntriesController.php
<?php
declare(strict_types=1);

namespace Luma\Controllers;

use Luma\Http\Response;
use PDO;

final class EntriesController
{
    public function __construct(private readonly PDO $pdo)
    {
    }

    public function create(): void
    {
        $body = requestBody();
        $entry = [
            'Id' => uuidV4(),
            'UserId' => getUserId(),
            'Content' => (string)($body['content'] ?? $body['Content'] ?? ''),
            'Mood' => (int)($body['mood'] ?? $body['Mood'] ?? 0),
            'CreatedAtUtc' => utcNow(),
            'UpdatedAtUtc' => null,
        ];

        $statement = $this->pdo->prepare(
            'INSERT INTO "JournalEntries" ("Id", "UserId", "Content", "Mood", "CreatedAtUtc", "UpdatedAtUtc")
             VALUES (:Id, :UserId, :Content, :Mood, :CreatedAtUtc, :UpdatedAtUtc)'
        );
        $statement->execute($entry);

        Response::json($entry, 201, ['Location' => '/api/v1/entries/' . $entry['Id']]);
    }

    public function all(): void
    {
        $statement = $this->pdo->prepare(
            'SELECT "Id", "UserId", "Content", "Mood", "CreatedAtUtc", "UpdatedAtUtc"
             FROM "JournalEntries"
             WHERE "UserId" = :userId
             ORDER BY "CreatedAtUtc" DESC
             LIMIT 50'
        );
        $statement->execute(['userId' => getUserId()]);

        Response::json($statement->fetchAll());
    }

    public function find(string $id): void
    {
        $entry = $this->findEntry($id);
        $entry ? Response::json($entry) : Response::json(null, 404);
    }

    public function update(string $id): void
    {
        $entry = $this->findEntry($id);
        if (!$entry) {
            Response::json(null, 404);
            return;
        }

        $body = requestBody();
        $entry['Content'] = (string)($body['content'] ?? $body['Content'] ?? $entry['Content']);
        $entry['Mood'] = (int)($body['mood'] ?? $body['Mood'] ?? $entry['Mood']);
        $entry['UpdatedAtUtc'] = utcNow();

        $statement = $this->pdo->prepare(
            'UPDATE "JournalEntries"
             SET "Content" = :Content, "Mood" = :Mood, "UpdatedAtUtc" = :UpdatedAtUtc
             WHERE "Id" = :Id AND "UserId" = :UserId'
        );
        $statement->execute([
            'Content' => $entry['Content'],
            'Mood' => $entry['Mood'],
            'UpdatedAtUtc' => $entry['UpdatedAtUtc'],
            'Id' => $entry['Id'],
            'UserId' => $entry['UserId'],
        ]);

        Response::json($entry);
    }

    public function delete(string $id): void
    {
        $entry = $this->findEntry($id);
        if (!$entry) {
            Response::json(null, 404);
            return;
        }

        $statement = $this->pdo->prepare('DELETE FROM "JournalEntries" WHERE "Id" = :id AND "UserId" = :userId');
        $statement->execute(['id' => $id, 'userId' => getUserId()]);

        Response::json(null, 204);
    }

    private function findEntry(string $id): array|false
    {
        $statement = $this->pdo->prepare(
            'SELECT "Id", "UserId", "Content", "Mood", "CreatedAtUtc", "UpdatedAtUtc"
             FROM "JournalEntries"
             WHERE "Id" = :id AND "UserId" = :userId
             LIMIT 1'
        );
        $statement->execute(['id' => $id, 'userId' => getUserId()]);

        return $statement->fetch();
    }
}
